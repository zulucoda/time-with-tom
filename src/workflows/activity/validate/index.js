import activityConfig from '../constants';

/**
 * validate values
 * @function
 * @param {object} values - formik values
 * @returns {object} error - returns empty error object when no errors | returns error object
 */
const validate = (values) => {
  const errors = {};
  if (
    values[activityConfig.SELECT_NAME] === activityConfig.ACTIVITY_VALUE_NONE
  ) {
    errors[activityConfig.SELECT_NAME] = 'Please select activity';
  }
  return errors;
};

export default validate;
