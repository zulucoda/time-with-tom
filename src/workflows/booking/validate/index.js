import bookingConfig from '../constants';

/**
 * validate values
 * @function
 * @param {object} values - formik values
 * @returns {object} error - returns empty error object when no errors | returns error object
 */
const validate = (values) => {
  const errors = {};

  if (values[bookingConfig.INPUT_DATE] === '') {
    errors[bookingConfig.INPUT_DATE] = 'Date is required.';
  }

  if (values[bookingConfig.INPUT_START_TIME] === '') {
    errors[bookingConfig.INPUT_START_TIME] = 'Start time is required.';
  }

  if (values[bookingConfig.INPUT_END_TIME] === '') {
    errors[bookingConfig.INPUT_END_TIME] = 'End time is required.';
  }

  return errors;
};

export default validate;
