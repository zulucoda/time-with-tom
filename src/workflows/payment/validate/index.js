import paymentConfig from '../constants';

/**
 * validate values
 * @function
 * @param {object} values - formik values
 * @returns {object} error - returns empty error object when no errors | returns error object
 */
const validate = (values) => {
  const errors = {};

  if (values[paymentConfig.INPUT_CC_NAME] === '') {
    errors[paymentConfig.INPUT_CC_NAME] = 'Cardholder name is required.';
  }

  if (values[paymentConfig.INPUT_CC_NUMBER] === '') {
    errors[paymentConfig.INPUT_CC_NUMBER] = 'Card number is required.';
  }

  if (values[paymentConfig.INPUT_CC_EXPIRY_MONTH] === '') {
    errors[paymentConfig.INPUT_CC_EXPIRY_MONTH] =
      'Card expiry month is required.';
  }

  if (values[paymentConfig.INPUT_CC_EXPIRY_YEAR] === '') {
    errors[paymentConfig.INPUT_CC_EXPIRY_YEAR] =
      'Card expiry year is required.';
  }
  if (values[paymentConfig.INPUT_CC_CVC] === '') {
    errors[paymentConfig.INPUT_CC_CVC] = 'Card CVC is required.';
  }

  return errors;
};

export default validate;
