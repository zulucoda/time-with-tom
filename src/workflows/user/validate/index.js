import userConfig from '../constants';

/**
 * validate values
 * @function
 * @param {object} values - formik values
 * @returns {object} error - returns empty error object when no errors | returns error object
 */
const validate = (values) => {
  const errors = {};

  if (values[userConfig.INPUT_NAME] === '') {
    errors[userConfig.INPUT_NAME] = 'Name is required.';
  }

  if (values[userConfig.INPUT_SURNAME] === '') {
    errors[userConfig.INPUT_SURNAME] = 'Surname is required.';
  }

  if (values[userConfig.INPUT_EMAIL] === '') {
    errors[userConfig.INPUT_EMAIL] = 'Email is required.';
  }

  return errors;
};

export default validate;
