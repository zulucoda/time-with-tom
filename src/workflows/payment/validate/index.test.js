import validate from '.';
import paymentConfig from '../constants';

describe('Workflows - User - Validate - Unit Test', () => {
  test('when there are errors it should return error object', () => {
    const values = {
      [paymentConfig.INPUT_CC_NAME]: '',
      [paymentConfig.INPUT_CC_NUMBER]: '',
      [paymentConfig.INPUT_CC_EXPIRY_MONTH]: 0,
      [paymentConfig.INPUT_CC_EXPIRY_YEAR]: 0,
      [paymentConfig.INPUT_CC_CVC]: 0,
    };
    const act = validate(values);
    expect(act).toMatchSnapshot();
  });

  test('when there are NO errors it should return and empty error object', () => {
    const values = {
      [paymentConfig.INPUT_CC_NAME]: 'some cc name',
      [paymentConfig.INPUT_CC_NUMBER]: '55556666777788889999',
      [paymentConfig.INPUT_CC_EXPIRY_MONTH]: 1,
      [paymentConfig.INPUT_CC_EXPIRY_YEAR]: 2020,
      [paymentConfig.INPUT_CC_CVC]: 300,
    };
    const act = validate(values);
    expect(act).toMatchSnapshot();
  });
});
