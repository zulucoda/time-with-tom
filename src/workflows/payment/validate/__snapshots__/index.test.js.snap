// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflows - User - Validate - Unit Test when there are NO errors it should return and empty error object 1`] = `Object {}`;

exports[`Workflows - User - Validate - Unit Test when there are errors it should return error object 1`] = `
Object {
  "ccName": "Cardholder name is required.",
  "ccNumber": "Card number is required.",
}
`;
